src/rgw/rgw_rados.h

```diff
@@ -2810,18 +3052,21 @@ class RGWRados
   int bucket_suspended(rgw_bucket& bucket, bool *suspended);
 
   /** Delete an object.*/
-  virtual int delete_obj(RGWObjectCtx& obj_ctx,
-                         RGWBucketInfo& bucket_owner,
+  int delete_obj(RGWObjectCtx& obj_ctx,
+                         const RGWBucketInfo& bucket_owner,
                          const rgw_obj& src_obj,
                          int versioning_status,
                          uint16_t bilog_flags = 0,
                          const ceph::real_time& expiration_time = ceph::real_time());
 
+  /** Delete a raw object.*/
+  int delete_raw_obj(const rgw_raw_obj& obj);
+
```

Comment:
trying to separate the notion of a raw obj vs a system obj. Maybe this can be reworked later to avoid duplicate code though.